### ISSUE TITLE
Web UI access without user/password

### ISSUE THREAD

**Issue by pentago** [znc/znc#1913]
I'm using reverse proxy and would like to handle all my service authentication by using other methods, like basic HTTP auth or others. How can one disable user/password login form and go straight to the web UI?

**Comment by psychon** [comment]
Uhm... I think "use HTTP basic auth" is the answer, but since you are asking... uhm...

Just to confirm: Do I understand correctly that just using HTTP basic auth is not working for you?

**Comment by pentago** [comment]
No, I want to use HTTP basic auth but for it to make sense, I want to disable ZNC web UI login form and go straight to user page on page load.

**Comment by psychon** [comment]
I just tried it out and HTTP basic auth works here:

Without auth:
```
$ curl --silent http://localhost:1234 | grep 'Not logged in'       
			<span id="logged_in_no">Not logged in</span>
```
With auth:
```
$ curl --silent --basic --user abc:abc http://localhost:1234  | grep -E 'invalid|logged' 
			<span id="logged_in_yes">Logged in as: <span class="user_name">abc</span></span>
```
With invalid auth:
```
$ curl --silent --basic --user abc:abc2 http://localhost:1234  | grep -E 'invalid|logged'
<p>HTTP Basic authentication attempted with invalid credentials</p>
```
Checking that there is a login form without basic auth and no login form if I provided a valid basic auth login:
```
$ curl --silent --basic  http://localhost:1234  | grep form       
				<form action="/login" method="post">
				</form>
$ curl --silent --basic --user abc:abc http://localhost:1234  | grep form  
```

**Comment by pentago** [comment]
No, I think there's a misunderstanding.

What I'm looking for is to just disable ZNC web UI login form so I can use authentication on my reverse proxy level (Traefik+Authelia) and avoid having to log in twice.

**Comment by psychon** [comment]
Ah, sorry.

Uhm, how do you intend to tell apart users? The proxy does its own authentication and then adds some headers with the result? My only ideas right now would be "hardcode the password in the proxy so that it sends that along" and "implement some own module that doesn't exist yet so that no password is needed".

Sorry.

**Comment by pentago** [comment]
Ah okay. Consisering im the only user I have no need for authenticatin on each service i run. I know some projects like Calibre Web allow defining a header that proxy sends so it lets user in. Perhaps something like that would be great to have.
